edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -111,22 +99,23 @@ internal static string ValidateAndGetImage(string image, IEnvironmentWrapper env
         {
             image = Preconditions.CheckNonWhiteSpace(image, nameof(image)).Trim();
 
-            if (image.Substring(0, 1) == "$")
+            if ((image.Length > 0) && (image[0] == '$'))
```

Comment:
you don't need to check image.Length > 0 as Preconditions.CheckNonWhiteSpace already does.

Reply:
I think you can just use image.StartsWith for "$upstream" with string comparison to ignore case.  https://docs.microsoft.com/en-us/dotnet/api/system.string.startswith?view=netcore-3.1#System_String_StartsWith_System_String_System_StringComparison_

Reply:
I think it is better to support case-insensitive for $upstream.

Reply:
I agree for the first two. Not the last part. I'll rather be pedantic.